sequencer/src/l1_client.rs

```diff
@@ -129,6 +130,43 @@ impl L1Client {
             }
         }
     }
+    /// Get fee info for each `Deposit` ocurring between `prev`
+    /// and `new`. Returns `Vec<FeeInfo>`
+    pub async fn _get_finalized_deposits(
+        &self,
+        prev_finalized: Option<u64>,
+        new_finalized: u64,
+        address: Address,
+    ) -> Vec<FeeInfo> {
+        // `prev` should have allready been processed unless we
+        // haven't processed *any* blocks yet.
+        let prev = prev_finalized.map(|prev| prev + 1).unwrap_or(0);
+        // No new blocks have been finalized, therefore there are no
+        // new deposits.
+        if prev == new_finalized {
+            return vec![];
+        }
+
+        // query for deposit events, loop until successfull.
+        let events = loop {
+            match contract_bindings::fee_contract::FeeContract::new(
+                address,
+                Arc::new(&self.provider),
+            )
+            .deposit_filter()
+            .from_block(prev)
+            .to_block(new_finalized)
+            .query()
+            .await
+            {
+                Ok(events) => break events,
+                Err(e) => {
+                    tracing::warn!("Fee Event Error: {}", e);
```

Comment:
Add some delay between retries?

Reply:
How embarrassing. fixed in ff623fc0c816e46adcf629ad68c7f28dc024a75b